packages/data-context/src/sources/RelevantRunSpecsDataSource.ts

```diff
@@ -153,22 +139,38 @@ export class RelevantRunSpecsDataSource {
         statuses: {},
       }
 
-      if (cloudProject.current && cloudProject.current.runNumber && cloudProject.current.status) {
+      const { current, next } = cloudProject
+
+      if (
+        current
+         && current.runNumber
+         && current.status
+         && typeof current.totalInstanceCount === 'number'
+         && typeof current.completedInstanceCount === 'number'
+      ) {
         runSpecsToReturn.runSpecs.current = {
-          ...this.#calculateSpecMetadata(cloudProject.current.specs || []),
-          runNumber: cloudProject.current.runNumber,
+          totalSpecs: current.totalInstanceCount,
+          completedSpecs: current.completedInstanceCount,
+          runNumber: current.runNumber,
         }
 
-        runSpecsToReturn.statuses.current = cloudProject.current.status
+        runSpecsToReturn.statuses.current = current.status
       }
 
-      if (cloudProject.next && cloudProject.next.runNumber && cloudProject.next.status) {
+      if (
+        next
+        && next.runNumber
+        && next.status
+        && typeof next.totalInstanceCount === 'number'
+        && typeof next.completedInstanceCount === 'number'
+      ) {
         runSpecsToReturn.runSpecs.next = {
-          ...this.#calculateSpecMetadata(cloudProject.next.specs || []),
-          runNumber: cloudProject.next.runNumber,
+          totalSpecs: next.totalInstanceCount,
+          completedSpecs: next.completedInstanceCount,
+          runNumber: next.runNumber,
         }
 
-        runSpecsToReturn.statuses.next = cloudProject.next.status
+        runSpecsToReturn.statuses.next = next.status
       }
```

Comment:
This block is pretty much a duplicate of the one above, worth extracting to a fn?

Reply:
I was marginal on whether to do this or not, and we are about to be right back in here in the next sprint or two doing further modifications. So seems slightly easier on all sides to leave this logic repeated, make the diff super simple, and the shortly we'll combine them in a way that meets the needs we are about to have.

Reply:
I went ahead and made a helper: [`f138cc3` (#25757)](https://github.com/cypress-io/cypress/pull/25757/commits/f138cc379d606c1c286aaec0c59d44cb60d21356)